src/nano_contracts/deserializer.ts

```diff
@@ -24,45 +31,58 @@ class Deserializer {
    * We receive these types from the full node, so we
    * use the python syntax
    *
-   * @param {value} Value to deserialize
-   * @param {type} Type of the value to be deserialized
+   * @param {Buffer} buf Value to deserialize
+   * @param {string} type Type of the value to be deserialized
    *
    * @memberof Deserializer
    * @inner
    */
-  deserializeFromType(value: Buffer, type: string): NanoContractArgumentType | null {
-    if (type.endsWith('?')) {
-      // It's an optional
-      const optionalType = type.slice(0, -1);
-      return this.toOptional(value, optionalType);
-    }
-
-    if (type.startsWith('SignedData[')) {
-      return this.toSigned(value, type);
+  deserializeFromType(
+    buf: Buffer,
+    type: string
+  ): DeserializeResult<NanoContractArgumentType | null> {
+    const isContainerType = getContainerType(type) !== null;
+    if (isContainerType) {
+      return this.deserializeContainerType(buf, type);
     }
 
     switch (type) {
       case 'str':
-        return this.toString(value);
+        return this.toString(buf);
       case 'bytes':
-      case 'TxOutputScript':
-      case 'TokenUid':
+      case 'BlueprintId':
       case 'ContractId':
+      case 'TokenUid':
+      case 'TxOutputScript':
       case 'VertexId':
-        return this.toBytes(value);
+        return this.toBytes(buf);
       case 'Address':
-        return this.toAddress(value);
+        return this.toAddress(buf);
       case 'int':
       case 'Timestamp':
-        return this.toInt(value);
+        return this.toInt(buf);
       case 'Amount':
-        return this.toAmount(value);
-      case 'float':
-        return this.toFloat(value);
```

Comment:
chore: We should inform in the PR description that the `float` type is being removed: this is a breaking change, even though it only affects the alpha nano testnet.

Reply:
Done